docs/methods.md

```diff
@@ -1808,6 +1990,138 @@ url: 'ifps://QmYr1VJLwerg6pEoscdhVGugo39pa6rycEZLjtRPDfW84UAx'
 
 </details>
 
+### getDataFromExternalSources
+
+```js
+getDataFromExternalSources(schema, dataFromChain, ipfsGateway, [
+  (throwException = true),
+]);
+```
+
+Retrieve the data(s) stored on IPFS using one (or multiple) IPFS CID(s) and link(s) under the `dataFromChain` parameter.
+
+#### Parameters
+
+| Name                        | Type                 | Description                                                                                                                                    |
+| :-------------------------- | :------------------- | :--------------------------------------------------------------------------------------------------------------------------------------------- |
+| `schema`                    | `ERC725JSONSchema[]` | A list of JSON schemas for some given data keys.                                                                                               |
+| `dataFromChain`             | `DecodeDataOutput[]` | ...                                                                                                                                            |
+| `ipfsGateway`               | `string`             | The URL to use to fetch data from the IPFS network.                                                                                            |
+| `throwException` (optional) | `boolean`            | Define if the function should throw an error when fetching data. Set to `true` by default. This is useful for handling exceptions differently. |
+
+<details>
+    <summary>Types details</summary>
+
+```js
+interface ERC725JSONSchema {
+  name: string; // Describes the name of the key, SHOULD be composed of the Standards name + sub type. e.g: LSP2Name
+  key: string; // The keccak256 hash of the name. This is the actual key that MUST be retrievable via ERC725Y.getData(bytes32 key)
+  keyType: ERC725JSONSchemaKeyType | string; // Types that determine how the values should be interpreted.
+  valueContent: ERC725JSONSchemaValueContent | string; // string holds '0x1345ABCD...' If the value content are specific bytes, than the returned value is expected to equal those bytes.
+  valueType: ERC725JSONSchemaValueType | string; // The type of the value. This is used to determine how the value should be encoded / decode (`string` for tuples and CompactBytesArray).
+}
+
+interface DecodeDataOutput {
+  value: Data | Data[] | URLDataWithHash | null;
+  name: string;
+  key: string;
+}
+
+type Data = string | number | boolean | null;
+
+interface URLDataWithHash extends URLData {
+  verification: Verification; // | string is to allow use of string directly without importing the enum
+  json?: never;
+}
+
+interface URLData {
+  url: string;
+}
+```
+
+</details>
+
+#### Returns
+
+| Name | Type     | Description |
+| :--- | :------- | :---------- |
+|      | `object` | ...         |
```

Comment:
is it always an object? I think it can also return non JSON file no? Maybe a description would be good to add.